catalog/core/catalog-core-validationparser/src/main/java/ddf/catalog/validation/impl/ValidationParser.java

```diff
@@ -399,24 +402,61 @@ private void parseMetacardValidators(@Nullable Map<String, Object> root, Outer o
       Changeset changeset, Map<String, List<Outer.Validator>> validators) {
     List<Callable<Boolean>> staged = new ArrayList<>();
     for (Map.Entry<String, List<Outer.Validator>> entry : validators.entrySet()) {
-      Set<AttributeValidator> attributeValidators = validatorFactory(entry.getValue());
+      Set<ValidatorWrapper> validatorWrappers = validatorFactory(entry.getKey(), entry.getValue());
+      Set<AttributeValidator> attributeValidators =
+          validatorWrappers
+              .stream()
+              .map(ValidatorWrapper::getAttributeValidators)
+              .flatMap(Collection::stream)
+              .collect(Collectors.toSet());
       String attributeName = entry.getKey();
       staged.add(
           () -> {
             attributeValidatorRegistry.registerValidators(attributeName, attributeValidators);
             changeset.attributeValidators.put(attributeName, attributeValidators);
             return true;
           });
+      validatorWrappers
+          .stream()
+          .map(ValidatorWrapper::getMetacardValidators)
+          .flatMap(Collection::stream)
+          .collect(Collectors.toSet())
+          .forEach(
+              validator ->
+                  staged.add(
+                      () -> {
+                        ServiceRegistration<MetacardValidator> registration =
+                            getBundleContext()
+                                .registerService(MetacardValidator.class, validator, null);
+                        changeset.metacardValidatorServices.add(registration);
+                        return registration != null;
+                      }));
+      validatorWrappers
+          .stream()
+          .map(ValidatorWrapper::getReportingMetacardValidators)
+          .flatMap(Collection::stream)
+          .collect(Collectors.toSet())
+          .forEach(
+              validator ->
+                  staged.add(
+                      () -> {
+                        ServiceRegistration<ReportingMetacardValidator> registration =
+                            getBundleContext()
+                                .registerService(ReportingMetacardValidator.class, validator, null);
+                        changeset.reportingMetacardValidatorServices.add(registration);
+                        return registration != null;
+                      }));
     }
     return staged;
   }
 
-  private Set<AttributeValidator> validatorFactory(List<Outer.Validator> validators) {
+  private Set<ValidatorWrapper> validatorFactory(
+      String attribute, List<Outer.Validator> validators) {
     return validators
         .stream()
         .filter(Objects::nonNull)
         .filter(v -> StringUtils.isNotBlank(v.validator))
-        .map(this::getValidator)
+        .map((Outer.Validator key) -> getValidator(attribute, key))
```

Comment:
❓ you are calling `getValidator(attribute, key)`
and the method definition for get validator is `getValidator(String key, Outer.Validator validator) {`
Something is very confusing or attributes seem swapped.